src/machinery/useCallbackRef.js

```diff
@@ -0,0 +1,32 @@
+import { useDestroy } from "./useDestroy";
+import { noop } from "./utilities";
+
+/** 
+ * @param {{ onMount: Function, onUnmount: Function }} _
+ * @returns {import('react').RefObject<HTMLMediaElement>}
+ */
+export function useCallbackRef({ onMount = noop, onUnmount = noop }) {
```

Comment:
Deze lijkt overbodig, React snapt het ook als je functies meegeeft als ref.

Reply:
Ik kan me wel voorstellen dat je iets wilt hebben dat zich gedraagt als `useEffect`, maar dan in plaats van een dependency array een ref gebruikt.

Misschien iets als:

```js
const ref = useNodeRef(
  node => {
    ...
    () => {
       ...
    }
  },
  [...]
)
```
Dan gebruik je hetzelfde principe als `useEffect` die dit als signature heeft:

```
useEffect(install: () => UninstallFunction | void, dependencies: Array<any>)

useNodeRef(install: node => UninstallFunction | void, dependencies: Array<any>)

type UninstallFunction = () => void
```

Reply:
Check, ik heb deze opzet nu toegevoegd ✅